packages/toolpad-app/src/runtime/ToolpadApp.tsx

```diff
@@ -322,15 +322,18 @@ function RenderedNodeContent({ node, childNodeGroups, Component }: RenderedNodeC
     return {};
   }, [childNodeGroups.children, node]);
 
+  const components = useComponents();
+
   const props: Record<string, any> = React.useMemo(() => {
     return {
       ...boundProps,
       ...onChangeHandlers,
       ...eventHandlers,
       ...layoutElementProps,
       ...reactChildren,
+      __toolpadComponents: components,
```

Comment:
I'm in favor of adding a context

Reply:
React context or comment context? Can you ellaborate?

Reply:
Move the components context to `@mui/toolpad-core` and reuse instead of passing this prop

Reply:
Prop removed and replaced with context